scrypto/src/crypto/ecdsa.rs

```diff
@@ -1,60 +1,63 @@
+use p256::ecdsa::{signature::Signer, Signature, SigningKey};
+use p256::ecdsa::{signature::Verifier, VerifyingKey};
+use p256::elliptic_curve::sec1::{FromEncodedPoint, ToEncodedPoint};
+
+use p256::{EncodedPoint, PublicKey, SecretKey};
 use sbor::*;
 
-use crate::crypto::{Hash, HASH_LENGTH};
-use crate::misc::copy_u8_array;
 use crate::rust::borrow::ToOwned;
 use crate::rust::fmt;
 use crate::rust::str::FromStr;
 use crate::rust::string::String;
 use crate::rust::vec::Vec;
 use crate::types::{scrypto_type, ScryptoType};
 
-pub const ECDSA_PRIVATE_KEY_LENGTH: usize = 32;
-pub const ECDSA_PUBLIC_KEY_LENGTH: usize = 33;
-pub const ECDSA_SIGNATURE_LENGTH: usize = 65;
-
 /// Represents an ECDSA private key.
-#[derive(Clone, Copy, PartialEq, Eq, TypeId, Encode, Decode)]
-pub struct EcdsaPrivateKey(pub [u8; ECDSA_PRIVATE_KEY_LENGTH]);
+#[derive(Clone, PartialEq, Eq)]
+pub struct EcdsaPrivateKey(SecretKey);
```

Comment:
Do we really need `Clone`? I'm just being extra paranoid about not having secret keys everywhere.

Reply:
What's the issue? Even the `SecretKey` from 3rd-party supports Clone.

Reply:
Possibly harder to track where private key has been cloned/moved. But this is minor.